sdk/nodejs/resource.ts

```diff
@@ -334,7 +334,11 @@ export abstract class Resource {
             }
 
             // Update aliases to include the full set of aliases implied by the child and parent aliases.
-            opts.aliases = allAliases(opts.aliases || [], name, t, parent, parent.__name!);
+            monitorSupportsAliasSpecs().then(supportsAliasSpecs => {
```

Comment:
There is no way this is safe :| I think you need to pass the raw inputs down to registerResource et.al where it's safe to do async checks against the monitor.

Reply:
Alright let me be a little less obtuse :)